docs/user-guide/rug/event-handlers.md

```diff
@@ -0,0 +1,204 @@
+their own projects rather than in an existing project as the documentation describes
+for Rug generators or editors.
+
+{!handler-layout.md!}
+
+### Example Event Handler
+
+The following event handler responds when a tag is added to or deleted from an
+issue. When this event occurs, the Atomist Bot sends a message to the
+repository's Slack channel informing people in the channel about the tagging
+_event_.
+
+```typescript linenums="1"
+import { EventHandler, Tags } from "@atomist/rug/operations/Decorators";
+import { HandleEvent, LifecycleMessage, Plan } from "@atomist/rug/operations/Handlers";
+import { GraphNode, Match, PathExpression } from "@atomist/rug/tree/PathExpression";
+
+import { Comment } from "@atomist/cortex/Comment";
+import { Issue } from "@atomist/cortex/Issue";
+
+@EventHandler("ClosedGitHubIssues", "Handles closed issue events",
+    new PathExpression<Issue, Issue>(
+        `/Issue()[@state='closed']
+            [/resolvingCommits::Commit()/author::GitHubId()
+                [/person::Person()/chatId::ChatId()]?]?
+            [/openedBy::GitHubId()
+                [/person::Person()/chatId::ChatId()]?]
+            [/closedBy::GitHubId()
+                [/person::Person()/chatId::ChatId()]?]?
+            [/repo::Repo()/channels::ChatChannel()]
+            [/labels::Label()]?`))
+@Tags("github", "issue")
+class ClosedIssue implements HandleEvent<Issue, Issue> {
+    handle(event: Match<Issue, Issue>): Plan {
+        const issue = event.root();
+
+        const lifecycleId = "issue/" + issue.repo.owner + "/" + issue.repo.name + "/" + issue.number;
+        const message = new LifecycleMessage(issue, lifecycleId);
+
+        message.addAction({
+            label: "Reopen",
+            instruction: {
+                kind: "command",
+                name: "ReopenGitHubIssue",
+                parameters: {
+                    issue: issue.number,
+                    owner: issue.repo.owner,
+                    repo: issue.repo.name,
+                },
+            },
+        });
+
+        return Plan.ofMessage(message);
+    }
+}
+export const closedIssue = new ClosedIssue();
+```
+
+This event handler follows the same programming model as other Rugs, so it 
+should look familiar. It gets triggered for each GitHub repository `Tag` event
+in the team and responds by sending a [Lifecycle Message](#messages) to the Atomis Bot.
+
+### Declaration
+
+Declaring an event handler is done using the `#!typescript @EventHandler` class
+decorator, requires name and description. Additionally, it requires a 
+[Path Expression][pes], which registers the handler as handling new new events
+resulting from the expression execution.
+
+### Implementation
+
+Rug event handlers must implemented the `#!typescript HandleEvent<R,M>`, where
+the `R` and `M` type parameters refer to the expected root node and match types
+resulting from the execution of a [Path Expression][pes] respectively.
```

Comment:
"refer to the expected root node and match types, respectively, resulting from the execution"